src/h5web/visualizations/heatmap/HeatmapVis.tsx

```diff
@@ -33,8 +37,14 @@ function HeatmapVis(): JSX.Element {
               <ambientLight />
               {/* Provide context again - https://github.com/react-spring/react-three-fiber/issues/262 */}
               <HeatmapProvider {...props}>
-                <AxisGrid />
-                <Tooltip />
+                <AxisGrid
+                  // -0.5 to have ticks at the center of pixels
+                  abscissaDomain={[-0.5, cols - 0.5]}
+                  ordinateDomain={[-0.5, rows - 0.5]}
+                  axisOffsets={axisOffsets}
+                />
+                <TooltipMesh dims={dims} data={data} />
```

Comment:
You can revert to calling `useProps` inside `TooltipMesh` instead of passing props.

Reply:
I would like to keep it as it-is for now !